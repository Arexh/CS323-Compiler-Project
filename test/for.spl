int main(){
    int a;
    for(int a = 0; a < 100; a = a + 1){
        if(a == 50){
            break;
        }
    }
    return 0;
}