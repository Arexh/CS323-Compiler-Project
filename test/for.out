Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: main
        LP
        RP
      CompSt (1)
        LC
        DefList (2)
          Def (2)
            Specifier (2)
              TYPE: int
            DecList (2)
              Dec (2)
                VarDec (2)
                  ID: a
            SEMI
        StmtList (3)
          Stmt (3)
            FOR
            LP
            Def (3)
              Specifier (3)
                TYPE: int
              DecList (3)
                Dec (3)
                  VarDec (3)
                    ID: a
                  ASSIGN
                  Exp (3)
                    INT: 0
              SEMI
            Exp (3)
              Exp (3)
                ID: a
              LT
              Exp (3)
                INT: 100
            SEMI
            Exp (3)
              Exp (3)
                ID: a
              ASSIGN
              Exp (3)
                Exp (3)
                  ID: a
                PLUS
                Exp (3)
                  INT: 1
            RP
            Stmt (3)
              CompSt (3)
                LC
                StmtList (4)
                  Stmt (4)
                    IF
                    LP
                    Exp (4)
                      Exp (4)
                        ID: a
                      EQ
                      Exp (4)
                        INT: 50
                    RP
                    Stmt (4)
                      CompSt (4)
                        LC
                        StmtList (5)
                          Stmt (5)
                            Exp (5)
                              ID: break
                            SEMI
                        RC
                RC
          StmtList (8)
            Stmt (8)
              RETURN
              Exp (8)
                INT: 0
              SEMI
        RC
